File: Algorithm/Class/C C++ Python/Class 2/Test04/Test04.c
#include <stdio.h>

int main()
{
   int m ,n,temp;
   while(scanf("%d %d",&m,&n),m + n != -2) {

        while(n!=0){
            temp = m%n;
            m = n;
            n = temp;
        }
        if(m==1)
            printf("YES\n");
        else
            printf("POOR Haha\n");
   }
   return 0;
}

File: Algorithm/Class/Java/Class2/src/com/two/Test04.java
package com.two;

import java.util.Scanner;

public class Test04 {
	public static void main(String[] args) {
		Scanner scanner = new Scanner(System.in);
		while (scanner.hasNextInt()) {
			int m = scanner.nextInt();
			int n = scanner.nextInt();
			if (m == -1 && n == -1) {
				break;
			}
			int temp;
			if (n > m) {
				temp = m;
				m = n;
				n = temp;
			}
			int a = fun(m, n);
			if (a == 1) {
				System.out.println("YES");

			} else {
				System.out.println("POOR Haha");
			}
		}
	}

	public static int fun(int m, int n) {
		int a;
		if (m % n == 0) {
			return n;
		} else {
			a = m % n;
			return fun(n, a);
		}
	}
}
